include/envoy/ssl/context.h

```diff
@@ -40,6 +43,19 @@ class Context {
    * expire, or `absl::nullopt` if no OCSP responses exist.
    */
   virtual absl::optional<uint64_t> secondsUntilFirstOcspResponseExpires() const PURE;
+
+  /*
+   * Attempts to verify the certificate chain.
+   * @param leaf_cert the leaf certificate to be verified.
+   * @param intermediates the intermediate certificates, if present.
+   * @param error_details if an error is encountered, a human readable explanation will be stored
+   * here.
+   * @return true if the certificate chain is verified, false otherwise.
+   */
+  virtual bool verifyCertChain(x509_st& /*leaf_cert*/, stack_st_X509& /*intermediates*/,
+                               std::string& /*error_details*/) {
```

Comment:
Should this interface method also be PURE?

Reply:
I think it'd be a mess due to the openssl / boringssl requirement.  I'll add a comment.

Reply:
For that reason, I'm really not a fan of having boring/openssl types in the interface at all. Can we do something to hide the types here?

Reply:
I think the other option is to dynanmic cast the interface to the impl over in source/common/quic/envoy_quic_proof_verifier.cc  Want me to go that route?

Reply:
I think that's probably better. Especially because how these types is declared/defined in boring/openssl is unique.